### PR TITLE
Add Source Gen Logging for Worker Config

### LINKED ISSUE #9503: Log WorkerConfigOptions
We currently log a bunch of different options objects but not the WorkerConfig. It'd be good to see exactly what the worker is running with -- today we are fairly blind as customers can overwrite some values and we don't notice them.

See other Options that implement `IOptionsFormatter` to write these out.

### PR DESCRIPTION
**Pull request by aishwaryabh**
### Issue describing the changes in this PR

resolves [#9503](https://github.com/Azure/azure-functions-host/issues/9503)

This PR adds logging for the worker config using json source gen. I also checked in the benchmark tests I created for reference

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptions.cs [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+
+        public string Format()
+        {
+            return JsonSerializer.Serialize(this, typeof(LanguageWorkerOptions), SourceGenerationContextWorkerConfig.Default);
+        }
+    }
+
+    [JsonSourceGenerationOptions(WriteIndented = true, IncludeFields = false)]
+    [JsonSerializable(typeof(LanguageWorkerOptions))]
+    internal partial class SourceGenerationContextWorkerConfig : JsonSerializerContext
```
Comment (brettsam) [inline src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptions.cs]: Let's call this `LanguageWorkerOptionsJsonSerializerContext`...

Comment (brettsam) [reply src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptions.cs]: Oh and you can remove `IncludeFields = false` -- I did that as a test earlier.

File: src/WebJobs.Script/WebJobs.Script.csproj [hunk 2]
```diff
@@ -83,6 +83,7 @@
     <PackageReference Include="System.Reactive.Linq" Version="5.0.0" />
     <PackageReference Include="System.Reactive.Core" Version="5.0.0" />
     <PackageReference Include="System.Runtime.Loader" Version="4.3.0" />
+    <PackageReference Include="System.Text.Json" Version="6.0.9" />
```
Comment (brettsam) [inline src/WebJobs.Script/WebJobs.Script.csproj:93]: Can you add a comment with a note here? Something like:

```xml
<!--
Explicitly referencing System.Text.Json to get two bug fixes for the Source Generator. Without this, the 6.0.0 version 
of the Source Generator is used, which causes issues. This reference can be removed in future framework versions.

(and then add links to those two issues?) 
-->
```

File: benchmarks/WebJobs.Script.Benchmarks/SerializationBenchmarks.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (brettsam) [inline benchmarks/WebJobs.Script.Benchmarks/SerializationBenchmarks.cs]: You can back this out... we don't need it checked in.

File: src/WebJobs.Script/WebJobs.Script.csproj (hunk 2 above)
Comment (aishwaryabh) [reply src/WebJobs.Script/WebJobs.Script.csproj:93]: Added!

File: benchmarks/WebJobs.Script.Benchmarks/SerializationBenchmarks.cs (hunk 3 above)
Comment (aishwaryabh) [reply benchmarks/WebJobs.Script.Benchmarks/SerializationBenchmarks.cs]: Removed!

File: src/WebJobs.Script/WebJobs.Script.csproj [hunk 4]
```diff
@@ -83,6 +83,14 @@
     <PackageReference Include="System.Reactive.Linq" Version="5.0.0" />
     <PackageReference Include="System.Reactive.Core" Version="5.0.0" />
     <PackageReference Include="System.Runtime.Loader" Version="4.3.0" />
+	<!--
+    Explicitly referencing version 6.0.9 of System.Text.Json to get two bug fixes for the Source Generator. Without this, the 6.0.0 version 
+    of the Source Generator is used, which causes issues. This reference can be removed in future framework versions.
+
+    https://github.com/dotnet/runtime/issues/62354
+	https://github.com/dotnet/runtime/issues/62354
```
Comment (brettsam) [inline src/WebJobs.Script/WebJobs.Script.csproj]: Can you fix the spacing? looks off on the whole comment. Also one of these should be issue 62082

**Comment by aishwaryabh** [comment]
deps.json unit test:

```
 Error Message:
   IMPORTANT: The dependencies in WebHost have changed and MUST be reviewed before proceeding. Please follow up with brettsam, fabiocav or mathewc for approval.

Previous file: D:\a\_work\1\s\test\WebJobs.Script.Tests\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json
New file:      D:\a\_work\1\s\src\WebJobs.Script.WebHost\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json

  Changed:

  Removed:

  Added:
    - System.Text.Json.dll: 6.0.0.0/6.0.2523.51912

```

**Review by brettsam — APPROVED** [review]

**Review by kshyju — APPROVED** [review]

**Pull request merged**